MonoGame.Framework.Content.Pipeline/ExternalTool.cs

```diff
@@ -27,13 +34,47 @@ public static int Run(string command, string arguments)
             return result;
         }
 
+        public static void RestoreDotnetTool(string command, string toolName, string toolVersion, string path)
+        {
+            Directory.CreateDirectory(path);
+            var exe = CurrentPlatform.OS == OS.Windows ? "dotnet.exe" : "dotnet";
+            var dotnetRoot = Environment.GetEnvironmentVariable("DOTNET_ROOT");
+            if (!string.IsNullOrEmpty(dotnetRoot))
+            {
+                exe = Path.Combine(dotnetRoot, exe);
+            }
+            if (Run(exe, $"tool {command} {toolName} --version {toolVersion} --tool-path .", out string _, out string _,  workingDirectory: path) != 0)
+            {
+                // install the latest
+                Run(exe, $"tool {command} {toolName} --tool-path .", out _, out _,  workingDirectory: path);
+            }
+        }
+
+        public static void RestoreDotnetTools()
```

Comment:
So I moved things around a bit, the `RestoreDotnetTools` now gets called in the static constructor on first use, then not at all after that, so that should help with file/io :D. I also made these called private as they are not part of the public api. 

We already have a exception handler for tools that are not present or missing, which happens here https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework.Content.Pipeline/ExternalTool.cs#L46. 

Given the plan is to swtich out allot of this stuff into a new native content pipeline so/dll/dylib which statically links all the tools we use and presents a simple API we can leave this as is for now I think.

Reply:
Cool, the static constructor sounds good.